assets/js/hooks/auth_code.js

```diff
@@ -0,0 +1,111 @@
+
+    handleOnChange() {
+        this.inputs.forEach((input, index) => {
+            input.addEventListener("input", (event) => {
+                const value = event.target.value;
+                const nextInput = this.inputs[index + 1];
+                const pattern = input.getAttribute("pattern");
+
+                if (value.length > 1) {
+                    event.target.value = value.charAt(0);
+                    if (nextInput !== undefined && !this.form) {
+                        nextInput.focus();
+                    }
+                } else if (value.match(new RegExp(pattern, "gi"))) {
+                    if (nextInput !== undefined && !this.form) {
+                        nextInput.focus();
+                    }
+                } else {
+                    event.target.value = "";
+                }
+                event.preventDefault();
+
+                this.updateHiddenField();
+            });
+        });
+    },
+
+    handleOnKeydown() {
+        this.inputs.forEach((input, index) => {
+            input.addEventListener("keydown", (event) => {
+                const prevInput = this.inputs[index - 1];
+
+                if (event.key === "Backspace" && !this.form) {
+                    if (input.value === "") {
+                        if (prevInput !== undefined) {
+                            prevInput.value = "";
+                            prevInput.focus();
+                        }
+                    } else {
+                        input.value = "";
+                    }
+                    this.updateHiddenField();
+                    event.preventDefault();
+                }
+            });
+        });
+    },
+
+    handleOnPaste() {
+        this.inputs.forEach((input, index) => {
+            input.addEventListener("paste", (event) => {
+                const count = this.inputs.length - 1;
+                const lastInput = this.inputs[count];
+
+                const pastedValue = event.clipboardData.getData("Text");
+                const pattern = input.getAttribute("pattern");
+
+                let currentInput = index;
+                let pastedCount = 0;
+
+                for (let i = 0; i < pastedValue.length; i++) {
+                    const pastedCharacter = pastedValue.charAt(i);
+
+                    if (
+                        pastedCharacter.match(pattern) &&
+                        currentInput < this.inputs.length
+                    ) {
+                        this.inputs[currentInput].value = pastedCharacter;
+                        currentInput++;
+                        pastedCount++;
+                    }
+                }
+
+                const nextInput = this.inputs[currentInput];
+
+                if (nextInput !== undefined) {
+                    nextInput.focus();
+                } else {
+                    lastInput.focus();
+                }
+
+                this.updateHiddenField();
+
+                this.inputs.forEach((input) => {
```

Comment:
i'd like to see here single event bubbling - only for hidden input, not for chunks